### PR TITLE
Implementing Laura's manual retention times

### PR DESCRIPTION
**Pull request by wkumler**
This pull request implements Laura's manual retention times, as obtained from Skyline. There are several steps I envision in this process:

1. Add the Skyline .csv outputs and make sure they don't break existing code
2. Read in the new files and mung
3. Replace existing RTs as defined in the Ingalls_Standards_NEW.csv(?) file with these new manual ones

This code builds on the current state of the lintroll branch.

### CODE REVIEW

**Comment by wkumler** [comment]
The script is currently working with the new files! A few things came up during the process:

1. The "polarity" parameter has been deprecated, in favor of the more specific filename matching. Since the Skyline output that Laura provided has specific details about RT and m/z per file, we can subset based on filename rather than needing to fiddle with polarity things.
2. The above step also means that we should decide on a way to produce a single consensus MSMS spectrum from multiple files, as each compound now has separate MSMS spectra for each file in which it was observed.
3. The consensus MSMS spectrum should account for the fact that different files are based on different Mixes, and we should merge the Ingalls.standards information to only use the correct mixes when creating the consensus MSMS.
4. Using custom RT's dramatically improved our coverage, from ~25% initially to ~75% by my preliminary assessment. Woohoo!

**Comment by wkumler** [comment]
I'm still not sure what the issue with Git is, but it seems to be working okay on my ends still. 
f99968b was done using the code currently in the script, no problem at all.

**Comment by R-Lionheart** [comment]
I just resolved a mini merge conflict and I think we're ready to combine. I'm wondering if I can just blame Mac for Git not working on my end or if I have an issue with the way my file stream is set up. Either way I'm glad it's working for you and that it will probably work for Laura on her machine.

**Comment by wkumler** [comment]
Okay, the Git stuff seems to be working now - I had to specify both files wrt the working directory, and it complains if the folder hasn't been added to Git already but I think it'll work on other computers. @R-Lionheart can you give this branch a try?

**Comment by R-Lionheart** [comment]
Still getting the same issues on my end. I think git is just not happy with my local directory, it's handling branch changes just fine and reflecting that in the repo,  but it keeps saying there's nothing to commit.

**Comment by R-Lionheart** [comment]
Ok, I had to add one more line to add the tracked .mzML files and then it worked for me. I hope I didn't sync that folder to Github, it looks like I didn't but I'm little suspicious of it working. I'm going to update this branch, do you want to try it again on your end and see if everything is working correctly?

**Review by wkumler — COMMENTED** [review]
Hmmmmmm this seems like a bad idea. We definitely *don't* want to add the raw mzMLs to this repo, and especially not to GitHub. You think it's responsible for giving you errors when you try to commit the other folders? I'd be surprised but won't guarantee it's not a problem.

File: MSMS/scripts/extract_DDA.R [hunk 1]
```diff
@@ -7,8 +7,8 @@ library(tidyverse)
 ## Set absolute path to the Google Drive. The MS2 files live there and you must have your working directory 
 ## Ensure the getwd() command properly connects to the shared Drive from your filepath!
 
-setwd("G:\\Shared drives\\Ingalls Lab\\Collaborative_Projects\\Standards\\Ingalls_Standards\\MSMS") # Windows
-# setwd("/Volumes/GoogleDrive/Shared drives/Ingalls Lab/Collaborative_Projects/Standards/Ingalls_Standards/MSMS/") # Mac
+# setwd("G:\\Shared drives\\Ingalls Lab\\Collaborative_Projects\\Standards\\Ingalls_Standards\\MSMS") # Windows
+setwd("~/Google Drive/Shared drives/Ingalls Lab/Collaborative_Projects/Standards/Ingalls_Standards/MSMS/") # Mac
```
Comment (wkumler) [inline MSMS/scripts/extract_DDA.R:12]: This is a good idea! I changed the comment so that it's easiest for Laura, who's on Windows, to run this. I think we should make one decision and stick with it rather than committing back and forth repeatedly.

**Comment by R-Lionheart** [comment]
Ah, I was on the incorrect branch on the github desktop so it looked like it hadn't been added, but when I refreshed the page I see that the mzMLs were added. Yeah I'm definitely with you they shouldn't be synced. Adding them did fix the issues but this is not the way. Let me try again. I'm not concerned if they're working for you since that means they'll work for Laura.

**Comment by wkumler** [comment]
Great. I think it's ready to merge!

**Comment by R-Lionheart** [comment]
OK I actually fixed it this time, it's all working from my end! Merge whenever you're ready

File: MSMS/scripts/extract_DDA.R [hunk 2]
```diff
@@ -123,8 +124,8 @@ if (file.size("data_processed/Ingalls_Lab_Standards_MSMS.csv") / 1e6 > 5) {
   stop("Beware! The output file is larger than 5MB - be careful pushing to GitHub.")
 }
 
-library(git2r)
 
+# Automatically commit changes to the Ingalls Standards github
 repo <- repository()
 add(repo, ".gitignore.txt")
```
Comment (wkumler) [inline MSMS/scripts/extract_DDA.R]: What's the thought behind this line? .gitignore.txt doesn't actually exist as far as I can tell, and I'm nervous auto-adding a file that doesn't exist yet. I don't think the current .gitignore needs to be recommitted every time we update the MSMS data

Comment (R-Lionheart) [reply MSMS/scripts/extract_DDA.R]: I modified the gitignore to include all the .mzML files, which I think is just good practice since we never want those on the github. I'm seeing the gitignore on both the remote repo on GH and in my own repo, so I'm not sure what you mean by it doesn't exist?

**Comment by R-Lionheart** [comment]
Screw it on the error message, I just looked into trycatch and I don't want to mess with it. Before we fully finish/deploy it we should give Laura a heads up that if she gets that message its nbd.
